src/asyncopenstackclient/proxy.py

```diff
@@ -0,0 +1,31 @@
+class ResourceProxy:
+
+    def __init__(self, api, resource_name):
+        self.resource_name = resource_name
+        self.api = api
+
+    def __getattr__(self, name):
+        return MethodProxy(self.api, self.resource_name, name)
+
+
+class MethodProxy:
+
+    def __init__(self, api, resource, method):
+        self.api = api
+        self.resource = resource
+        self.method = method
+
+    def __call__(self, *args, **kwargs):
+        return self.__await__(*args, **kwargs)
+
+    def __await__(self, *args, **kwargs):
```

Comment:
IMHO `__await__` is a bit confusing, despite the code you can't do `await obj`

Reply:
shall we just move it to `__call__`?